### PR TITLE
Add commit and rollback nodes

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject

Part of https://github.com/wso2-enterprise/eggplant-product-management/issues/15

### CODE REVIEW

**Comment by KavinduZoysa** [comment]
Closing this as this is not required at the moment

**Comment by LakshanWeerasinghe** [comment]
Part of [implementing transaction handling flow nodes](https://github.com/wso2-enterprise/eggplant-project/issues/113).

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/AvailableNodesGenerator.java [hunk 1]
```diff
@@ -170,11 +170,14 @@ private void setDefaultNodes() {
                     .node(NodeKind.PANIC)
                     .stepOut();
         // TODO: Uncomment when concurrency is implemented
-//                .stepIn(Category.Name.CONCURRENCY)
-//                    .node(NodeKind.TRANSACTION)
+        this.rootBuilder
+                .stepIn(Category.Name.CONCURRENCY)
+                    .node(NodeKind.TRANSACTION)
+                    .node(NodeKind.COMMIT)
+                    .node(NodeKind.ROLLBACK)
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/AvailableNodesGenerator.java:177]: Let's disable these nodes, as the FE changes are yet to be implemented

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java [hunk 2]
```diff
@@ -110,34 +112,38 @@ public abstract class NodeBuilder {
...
-        put(NodeKind.STOP, Stop::new);
-        put(NodeKind.FUNCTION_CALL, FunctionCall::new);
-        put(NodeKind.FOREACH, Foreach::new);
-        put(NodeKind.DATA_MAPPER, DataMapper::new);
-        put(NodeKind.ASSIGN, Assign::new);
-        put(NodeKind.COMMENT, Comment::new);
-        put(NodeKind.MATCH, Match::new);
-    }};
+    private static final Map<NodeKind, Supplier<? extends NodeBuilder>> CONSTRUCTOR_MAP = new HashMap<>() {
+        {
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java:116]:
```suggestion
    private static final Map<NodeKind, Supplier<? extends NodeBuilder>> CONSTRUCTOR_MAP = new HashMap<>() {{
```

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/node/Commit.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import io.ballerina.flowmodelgenerator.core.model.NodeKind;
+import io.ballerina.flowmodelgenerator.core.model.SourceBuilder;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents the properties of a break node.
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/node/Commit.java:33]:
```suggestion
 * Represents the properties of a commit node.
```

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/node/Commit.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            sourceBuilder.token().keyword(SyntaxKind.CHECK_KEYWORD);
+        }
+
+        return sourceBuilder
+                .token()
+                .keyword(SyntaxKind.COMMIT_KEYWORD)
+                .endOfStatement()
+                .stepOut()
+                .textEdit(false).build();
+
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/node/Commit.java:64]:
```suggestion
```

**Review by nipunayf — APPROVED** [review]

**Pull request merged**